lib/boxr/auth.rb

```diff
@@ -53,6 +55,33 @@ def self.revoke_tokens(token, client_id: ENV['BOX_CLIENT_ID'], client_secret: EN
     auth_post(uri, body)
   end
 
+  # Exchange a fully-scoped token for one that is down-scoped to specific permissions,
+  # defined by the scopes passed in.  You can optionally restrict the token to a
+  # particular file.
+  #
+  # @param [String] subject_token The fully-scoped token to exchange
+  #
+  # @param [Array<String>] scopes A list of the scopes to allow in the new token
+  #
+  # @param [Integer, String] file (Optional) The ID of a file resource to restrict
+  #   the token to
+  # @return [BoxrMash]
+  def self.exchange_token(subject_token, scopes:, file: nil)
+    resource_url = ( file.present? ) ? "#{Boxr::Client::FILES_URI}/#{file}" : nil
+    uri = "https://api.box.com/oauth2/token"
+
+    params = {
+      subject_token: subject_token,
+      subject_token_type: TOKEN_EXCHANGE_TOKEN_TYPE,
+      scope: scopes.join(' '),
+      grant_type: TOKEN_EXCHANGE_GRANT_TYPE
+    }
+
+    params[:resource] = resource_url unless resource_url.nil?
+
+    auth_post(uri, params.to_query)
```

Comment:
Maybe this is a silly question,, but where does `to_query` come from? As far as I can tell, it's an ActiveSupport extension which this library does have as a dependency. Perhaps you were able to use this in your app because you're using rails? Either way, this will probably have to go in favor of the way the query is built above.